### ISSUE TITLE
Complex case - need help

### ISSUE THREAD

**Issue by Twisterking** [cult-of-coders/grapher#224]
So ... I have the following data structure:
**items collection** - sample item:
```
_id: 'itemBar',
item_name: 'Barname'
```
then there is a **openorders collection** consisting of listitems:
```
_id: 'openorderitemXY',
item_amount: 3, // item 'itemBar' times 3
itemId: 'itemBar' // linked to the `item` above via Grapher
```
and then there is a **orders collection** for already sent/done orders:
```
_id: 'orderitemYZ',
item_amount: 5,
item: {
  _id: 'itemBar',
  item_name: 'Barname'
}
```
why is `orders` not also using the same `itemId` logic as `openorders`? Because the `items` collection is constantly changing. So I want the already orders items to be "freezed" in the database and not be dependent on the `items` collection. makes sense?! 😄 

--> **Problem:**
Let's say I have an old order opened. which now lists some orderitems. I might wanna order the same item again I used to order some days ago. So I might wanna **order `itemBar` again**:

--> I need a link which basically does the following:
give me the `openorderitem._id`(in the example this is `openorderitemXY`) from the `item._id` (`itemBar`) inside of `orderitemYZ` so I can change the `item_amount` of `openorderitemXY` on the client.

so basically: `orderitemYZ` -> `item._id `is `itemBar`-> get openorderitem where `itemId` is also `itemBar`- should result in `openorderitemXY`.

I hope this makes sense. tried my best to explain! 😁 

Maybe you can help @theodorDiaconu! 😃 

If this is not yet possible, maybe I can provide a function instead of a string to addLinks' `field` attribute?!

**Comment by theodorDiaconu** [comment]
@Twisterking have you looked into meta links ? I think that is what you want. I had to read your issue 5 times to get a grasp of it, and I'm still unsure I understood it perfectly. Maybe a diagram would help

**Comment by Twisterking** [comment]
I think I got it working! 😄 

In `db/openorders/links.js`:
```
OrdersBody.addLinks({
  'currentItem': { // renamed so it is not colliding with my `item` attr
    type: 'one',
    collection: Items,
    field: 'item',
    metadata: true // look for the _id to "join" on at `item._id`
  }
})
```
to query: something like:
```
OrdersBody.createQuery({
  _id: 1,
  list_id: 1,
  item_amount: 1,
  currentItem: {
    item_ref: 1,
    item_name: 1,
    units: 1,
    unit: 1,
    openOrderItem: { // this is my inversed link to get the actual openorder item
      $filter({filters, options, params}) {
        if(params.currentOpenOrderId) filters.list_id = params.currentOpenOrderId;
      },
      _id: 1,
      item_amount: 1
    }
  }
});
```

Maybe not pretty and gotta do some pretty heavy "normalizing" afterwards as the data is so nested but whatever ... if it works ... 😃

**Comment by theodorDiaconu** [comment]
I'm glad you found a solution. I'm sorry I couldn't be of any more help. Maybe these type of questions are rather better suited for the Meteor Forums.
